tests/slist_ut/slist_ut.c

```diff
@@ -0,0 +1,1114 @@
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+#include <stdbool.h>
+#include <stdlib.h>
+
+#include "macro_utils/macro_utils.h"
+#include "umock_c/umock_c.h"
+#include "umock_c/umocktypes_bool.h"
+#include "umock_c/umocktypes_charptr.h"
+
+
+#include "c_util/slist.h"
+#include "c_util/containing_record.h"
+#include "testrunnerswitcher.h"
+
+typedef struct simpleItem_tag
+{
+    unsigned char index;
+    SINGLYLINKEDLIST_ENTRY link;
+} simpleItem, * pSimpleItem;
+
+#define TEST_CONTEXT ((const void*)0x0101)
+
+#define ENABLE_MOCKS
+
+MOCK_FUNCTION_WITH_CODE(, bool, test_match_function, PSINGLYLINKEDLIST_ENTRY, list_entry, const void*, match_context)
+MOCK_FUNCTION_END(true);
+
+MOCK_FUNCTION_WITH_CODE(, bool, test_condition_function, PSINGLYLINKEDLIST_ENTRY, list_entry, const void*, action_context, bool*, continueProcessing);
+*continueProcessing = true;
+MOCK_FUNCTION_END(true);
+
+MOCK_FUNCTION_WITH_CODE(, bool, test_condition_function_2, PSINGLYLINKEDLIST_ENTRY, list_entry, const void*, action_context, bool*, continueProcessing);
+simpleItem* item = CONTAINING_RECORD(list_entry, simpleItem, link);
+*continueProcessing = item->index != 3;
+MOCK_FUNCTION_END(true);
+
+MOCK_FUNCTION_WITH_CODE(, int, test_action_function, PSINGLYLINKEDLIST_ENTRY, list_entry, const void*, action_context, bool*, continueProcessing);
+*continueProcessing = true;
+MOCK_FUNCTION_END(0);
+
+MOCK_FUNCTION_WITH_CODE(, int, test_action_function_2, PSINGLYLINKEDLIST_ENTRY, list_entry, const void*, action_context, bool*, continueProcessing);
+simpleItem* item = CONTAINING_RECORD(list_entry, simpleItem, link);
+*continueProcessing = item->index != 3;
+MOCK_FUNCTION_END(0);
+
+MOCK_FUNCTION_WITH_CODE(, int, test_action_function_fail, PSINGLYLINKEDLIST_ENTRY, list_entry, const void*, action_context, bool*, continueProcessing);
+*continueProcessing = true;
+simpleItem* item = CONTAINING_RECORD(list_entry, simpleItem, link);
+(int)item->index;
+if (item->index == 3)
+{
+    return MU_FAILURE;
+}
+return 0;
+MOCK_FUNCTION_END();
```

Comment:
This should have the return in MOCK_FUNCTION_END

Reply:
Just to make it neat (there might be cases in the future maybe where the MOCK_FUNCTION_END would expand to more code than just return. If you call return in the middle of the MOCK_FUNCTION_WITH_CODE then that code would be bypassed.

Reply:
done